### ISSUE TITLE
Docker build not working (FunctionInfo.h missing)

### ISSUE THREAD

**Issue by menzow** [facebook/bistro#18]
Hey,

I just tried compiling bistro through Docker and was not able to complete the process. Somewhere during the build the compiler throws an error that `FunctionInfo.h` is missing from the `fbthrift` dependency. 

After digging through the container I was not able to recover log file with the error message. It would be great to get some info as to where these messages get stored. Running `find /home -type f -name *.log -exec /bin/bash -c "cat {} | grep FunctionInfo"` did not return anything useful.

I did manage to find a probable cause for this issue by searching for references to `FunctionInfo.h` in the source files (`/home/*`) and working backwards from there. The `Makefile.am` in the fbthrift repository includes only the transport/core/TransportRoutingHandler.h and transport/core/ThriftProcessor.h headers. My guess is that `FunctionInfo.h` should also be included here.  

https://github.com/facebook/fbthrift/blob/76d376e1b7d0f189708ef6438abb40862be360d5/thrift/lib/cpp2/Makefile.am#L182-L184

**Comment by menzow** [comment]
Also placed the issue in https://github.com/facebook/fbthrift/issues/229 since it seemed more relevant there.

To initiate the build I used the example given in this issue: https://github.com/facebook/bistro/issues/12#issuecomment-301454554
```
cd bistro
export os_image=ubuntu:16.04
export gcc_version=5
make_parallelism=2 ./build/fbcode_builder/travis_docker_build.sh
```

Thanks :)

**Comment by snarkmaster** [comment]
Try using this for the C++ build, for the next few days: https://github.com/facebook/bistro/commit/044cd9f5985eddc330727e59d16686499a2b0ccf

There are no substantive changes since then, and the hash I linked will build.

Apologies for the breakage, and thanks for reporting it. This week the Thrift build was being migrated from automake to CMake, and we did not manage to keep both builds working. I would expect things to go back to normal early next week. 

As far as the getting logs, the README recommends a fancier version of this:

```
.../travis_docker_build.sh &> build.log & tail -f build.log
``` 

In other words, the docker build makes no attempt to record logs automatically, it's kind of up to the user :)

**Comment by menzow** [comment]
Hey @snarkmaster ,

Thanks for your reply! Checking out #044cd9f indeed resolved the missing header issues.

**Comment by snarkmaster** [comment]
@menzow, the test failures you're seeing are almost certainly due to flaky tests — there are no known associated production bugs due to these failures. I do intend to clean these up at some point in the next 1-2 months, though. Thanks for your vigilance.

**Comment by snarkmaster** [comment]
I'll close this out since the issue title is not related to flaky tests, and the tests are off. Feel free to create a new issue for tests if you'd like an update when they are fixed.
